### DISCUSSION TITLE
拉取最新的 latest 镜像重启容器后，数据会被清空么

### DISCUSSION THREAD

**Discussion by L-cw** [usememos/memos#1510] [Q&A] [answered]
请教一下 docker 更新镜像重启容器后会把挂载的数据清空吗，因为一直有更新提示，就升级了，现在写了一个月的数据没了，看目录下也空了。各位大佬，这还有救不，docker 配置如下
```version: "3.0"
services:
  memos:
    image: neosmemo/memos:latest
    container_name: memos
    volumes:
      - ~/.memos/:/service/memos/data
    ports:
      - 5230:5230
 ```
操作是 
docker pull neosmemo/memos:latest
docker-compose stop memos
docker-compose up -d

**Comment by boojack** [comment]
可以进入到 `~/.memos/` 目录里，看看是否存在 `memos_prod.db` 文件。如果存在的话，可以再试试启动服务，建议直接通过 docker run 启动。

```
docker run -d --name memos -p 5230:5230 -v ~/.memos/:/var/opt/memos neosmemo/memos:latest
```

**Reply by L-cw** [reply, marked as answer]
啊我知道了，是我之前的数据卷挂错了，数据没能持久化到，我的问题😭感谢大佬回答
